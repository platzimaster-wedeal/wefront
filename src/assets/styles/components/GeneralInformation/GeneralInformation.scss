@import "../../partials/colors.scss";
@import "../../partials/fonts.scss";
@import "../../partials/shadows.scss";
@import "../../partials/text.scss";
@import "../../partials/sizes.scss";

.GeneralInformation__container {
 position: relative;
 width: 100%;
 padding: $space-size-medium;
 margin: $space-size-medium 0;
 background: #fff;
 border-radius: 12px;
 box-shadow: $shadow-small;

 figure {
   margin: 0;
   margin-right: $space-size-small;
   padding: 0;
   position: relative;
 }
}

.item--icon {
  color: $color-primary;
  @include font-size(30px);

 }

 .button-actionSA {
  position: absolute;
  right: 0px;
  top: 0;
  cursor: pointer;
 }

.GeneralInformation__title {
 @include font-size($font-size-normal);
 font-weight: 500;
 color: $color-primary;

 @include media(tablet) {
  @include font-size($font-size-big);
 }
}

hr {
  border: 1px solid $color-primary !important;
  border-radius: 3px !important;
  margin: $space-size-small 0 !important;
  @include media(tablet){
    border: 2px solid $color-primary !important;
  }
 
}

.GeneralInformation__item {
 display: flex;
 align-items: center;
 margin-bottom: $space-size-small;

 &:last-child {
  margin-bottom: 0;
 }
}

.item--text {
 color: $color-black-60;
 font-style: normal;
 font-weight: 300;
 letter-spacing: 0.02em;
 @include font-size($font-size-normal);

}
