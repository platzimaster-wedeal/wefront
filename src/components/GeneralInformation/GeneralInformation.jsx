import React from "react";
import { MdPerson } from "react-icons/md";
import { MdGroup } from "react-icons/md";
import { MdHome } from "react-icons/md";
import { MdModeEdit } from "react-icons/md";
import SmallButtonSA from "../Buttons/SmallButtonSA";

import "../../assets/styles/components/GeneralInformation/GeneralInformation.scss";

const GeneralInformation = ({ name, followers, location, auth = false }) => {
 const handleEditInfo = () => alert("Voy a editar");

 return (
  <div className="GeneralInformation__container">
   <h3 className="GeneralInformation__title">General Information</h3>
   <hr />
   <div className="GeneralInformation__item">
    <figure>
     {auth ? (
      <SmallButtonSA
       icon={MdModeEdit}
       width="20px"
       height="20px"
       fontSize="15px"
       onClick={handleEditInfo}
      />
     ) : null}

     <MdPerson className="item--icon" />
    </figure>
    <p className="item--text">{name}</p>
   </div>
   <div className="GeneralInformation__item">
    <figure>
     <MdGroup className="item--icon" />
    </figure>

    <p className="item--text">{followers} Followers</p>
   </div>
   <div className="GeneralInformation__item">
    <figure>
     {auth ? (
      <SmallButtonSA
       icon={MdModeEdit}
       width="20px"
       height="20px"
       fontSize="15px"
      />
     ) : null}
     <MdHome className="item--icon" />
    </figure>
    <p className="item--text">{location}</p>
   </div>
  </div>
 );
};

export default GeneralInformation;
