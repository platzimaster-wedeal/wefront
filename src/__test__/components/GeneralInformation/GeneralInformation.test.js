import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import GeneralInformation from "../../../components/GeneralInformation/GeneralInformation";

describe("<Button/>", () => {
 const generalInformation = mount(<GeneralInformation/>);
 test("Render del componente GeneralInformation", () => {
  expect(generalInformation.length).toEqual(1);
 });
 test("Render del tÃ­tulo de un componente", () => {
  expect(generalInformation.find(".GeneralInformation__title").text()).toEqual("General Information");
 });
});

describe("Button Snapshot", () => {
 test("Comprobar la UI del componente GeneralInformation", () => {
  const generalInformation = create(<GeneralInformation />);
  expect(generalInformation.toJSON()).toMatchSnapshot();
 });
});