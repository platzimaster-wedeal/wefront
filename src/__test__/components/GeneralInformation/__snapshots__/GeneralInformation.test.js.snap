// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Comprobar la UI del componente Button 1`] = `
<div
  className="GeneralInformation__container"
>
  <h3
    className="GeneralInformation__title"
  >
    General Information
  </h3>
  <hr />
  <div
    className="GeneralInformation__item"
  >
    <figure>
      <svg
        className="item--icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </figure>
    <p
      className="item--text"
    />
  </div>
  <div
    className="GeneralInformation__item"
  >
    <figure>
      <svg
        className="item--icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
    </figure>
    <p
      className="item--text"
    >
       Followers
    </p>
  </div>
  <div
    className="GeneralInformation__item"
  >
    <figure>
      <svg
        className="item--icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </figure>
    <p
      className="item--text"
    />
  </div>
</div>
`;

exports[`Button Snapshot Comprobar la UI del componente GeneralInformation 1`] = `
<div
  className="GeneralInformation__container"
>
  <h3
    className="GeneralInformation__title"
  >
    General Information
  </h3>
  <hr />
  <div
    className="GeneralInformation__item"
  >
    <figure>
      <svg
        className="item--icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </figure>
    <p
      className="item--text"
    />
  </div>
  <div
    className="GeneralInformation__item"
  >
    <figure>
      <svg
        className="item--icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
    </figure>
    <p
      className="item--text"
    >
       Followers
    </p>
  </div>
  <div
    className="GeneralInformation__item"
  >
    <figure>
      <svg
        className="item--icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </figure>
    <p
      className="item--text"
    />
  </div>
</div>
`;
